src/hotspot/share/compiler/disassembler.cpp

```diff
@@ -804,48 +815,23 @@ bool Disassembler::load_library(outputStream* st) {
   // 4. hsdis-<arch>.so  (using LD_LIBRARY_PATH)
   if (jvm_offset >= 0) {
     // 1. <home>/lib/<vm>/libhsdis-<arch>.so
-    if (jvm_offset + strlen(hsdis_library_name) + strlen(os::dll_file_extension()) < JVM_MAXPATHLEN) {
-      strcpy(&buf[jvm_offset], hsdis_library_name);
-      strcat(&buf[jvm_offset], os::dll_file_extension());
-      if (Verbose) st->print_cr("Trying to load: %s", buf);
-      _library = os::dll_load(buf, ebuf, sizeof ebuf);
-    } else {
-      if (Verbose) st->print_cr("Try to load hsdis library failed: the length of path is beyond the OS limit");
-    }
+    _library = dll_load(buf, jvm_offset, sizeof buf, ebuf, sizeof ebuf, st);
```

Comment:
All of `dll_load()` are called with incorrect arguments order - length is should be just after offset.

Reply:
My bad, Thanks for pointing out.